lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3707,3 +3733,75 @@ def get_test_data(delta=0.05):
     Y = Y * 10
     Z = Z * 500
     return X, Y, Z
+
+
+class _Quaternion:
+    """
+    Quaternions
+    consisting of scalar, along 1, and vector, with components along i, j, k
+    """
+
+    def __init__(self, scalar, vector):
+        self.scalar = scalar
+        self.vector = np.array(vector)
+
+    def __neg__(self):
```

Comment:
What is the use of this operator/method? It's not an operation really used for rotation quaternions afaik, as q == -q.

Reply:
I didn't have it initially, indeed it isn't really needed for the purposes of `Axes3D`. But it turns out it is nice to have for formulating the quaternion test results. That's the use.

Reply:
Makes sense to me